### ISSUE TITLE
autocorrecting with Style/RedundantCondition can eat comments

### ISSUE THREAD

**Issue by maxnotarangelo** [rubocop/rubocop#13199]
When autocorrecting with the `Style/RedundantCondition` cop, comments can get deleted. Here's an offending code snippet:

```ruby
if existing_data
  # Important note! Don't delete me!
  arr << existing_data
  # Another important note!
else
  arr << [hsh[:first_value], get_second_value(hsh)]
end
```

## Expected behavior

It's probably best not to autocorrect in this case.

## Actual behavior

```ruby
arr << existing_data || [hsh[:first_value], get_second_value(hsh)]
```

There's also a `Style/ConditionalAssignment` warning on this code snippet. I made sure that `Style/RedundantCondition` was the issue; if you disable the `Style/ConditionalAssignment` cop, you get the same behavior. And if you disable `Style/RedundantCondition` (with `Style/ConditionalAssignment` enabled), you get the following autocorrected code, with the comments still present:

```ruby
arr << if existing_data
         # Important note! Don't delete me!
         existing_data
       # Another important note!
       else
         [hsh[:first_value], get_second_value(hsh)]
       end
```

## RuboCop version

```
$ bundle exec rubocop -V
1.66.1 (using Parser 3.3.5.0, rubocop-ast 1.32.3, running on ruby 3.1.1) [arm64-darwin21]
```
